src/hooks/index.js

```diff
@@ -0,0 +1,97 @@
+import api from "../api/api";
+import { Server } from "../utils/config";
+import { useEffect, useReducer } from "react";
+
+export const FetchState = {
+  FETCH_INIT: 0,
+  FETCH_SUCCESS: 1,
+  FETCH_FAILURE: 2,
+};
+
+export const useGetTodos = (stale) => {
+  const reducer = (state, action) => {
+    switch (action.type) {
+      case FetchState.FETCH_INIT:
+        return { ...state, isLoading: true, isError: false };
+      case FetchState.FETCH_SUCCESS:
+        return {
+          ...state,
+          isLoading: false,
+          isError: false,
+          todos: action.payload,
+        };
+      case FetchState.FETCH_FAILURE:
+        return { ...state, isLoading: false, isError: true };
+      default:
+        throw new Error();
+    }
+  };
+
+  const [state, dispatch] = useReducer(reducer, {
+    isLoading: false,
+    isError: false,
+    todos: [],
+  });
+
+  useEffect(() => {
+    let didCancel = false;
+    const getTodos = async () => {
+      dispatch({ type: FetchState.FETCH_INIT });
+      try {
+        const data = await api.listDocuments(Server.collectionID);
+        if (!didCancel)
+          dispatch({ type: FetchState.FETCH_SUCCESS, payload: data.documents });
+      } catch (e) {
+        if (!didCancel) dispatch({ type: FetchState.FETCH_FAILURE });
+      }
+    };
+    getTodos();
```

Comment:
Can't see a reason for this:
```js
const foo = () => {```}
foo();
```

You can do:
```js
  (async() => {
    await action();
  })()
```

Reply:
makes sense 😄

Reply:
I was refactoring this and went through it again, and I felt I should leave the syntax as is because the concise version that you provided may not be very obvious to new devs. WDYT 🤔 ?